File: journeys-backend/models/placereview.go
package models

import "time"

type PlaceReview struct {
	ReviewID  uint               `gorm:"primaryKey;column:review_id" json:"review_id"`
	UserID    uint               `gorm:"column:user_id" json:"user_id"`
	RouteID   uint               `gorm:"column:route_id" json:"route_id"`
	Rating    int                `gorm:"column:rating" json:"rating"`
	Comment   string             `gorm:"column:comment" json:"comment"`
	CreatedAt time.Time          `gorm:"column:created_at" json:"created_at"`
	Image     []PlaceReviewImage `gorm:"foreignKey:ReviewID;references:ReviewID" json:"image"`
}

type PlaceReviewImage struct {
	ImageID  uint   `gorm:"primaryKey;column:image_id" json:"image_id"`
	ReviewID uint   `gorm:"column:review_id" json:"review_id"`
	Image    []byte `gorm:"column:image" json:"image"`
}

File: journeys-frontend/lib/models/place_review.dart
class PlaceReview {
  final int reviewId;
  final int userId;
  final int routeId;
  final int rating;
  final String comment;
  final List<String> imageBase64List; // <- gunakan ini
  final DateTime createdAt;

  PlaceReview({
    required this.reviewId,
    required this.userId,
    required this.routeId,
    required this.rating,
    required this.comment,
    required this.imageBase64List,
    required this.createdAt,
  });

  factory PlaceReview.fromJson(Map<String, dynamic> json) {
    final image = (json['image'] as List<dynamic>?)
        ?.map((img) => img.toString())
        .toList() ?? [];

    return PlaceReview(
      reviewId: json['review_id'],
      userId: json['user_id'],
      routeId: json['route_id'],
      rating: json['rating'],
      comment: json['comment'],
      imageBase64List: image,
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}
